Source/WebGPU/WebGPU/Device.mm

```diff
@@ -510,6 +510,9 @@ static void setOwnerWithIdentity(id<MTLResourceSPI> resource, auto webProcessID)
 
 id<MTLTexture> Device::newTextureWithDescriptor(MTLTextureDescriptor *textureDescriptor, IOSurfaceRef ioSurface, NSUInteger plane) const
 {
+#if PLATFORM(MAC) || PLATFORM(MACCATALYST)
+    textureDescriptor.storageMode = MTLStorageModeManaged;
+#endif
```

Comment:
Is it correct to use `MTLStorageModeManaged` for macCatalyst builds? From reading the relevant validation code I thought you'd need to use `MTLStorageModeShared`, like on iOS.

Also: why override the `storageMode` here instead of setting it correctly at the different call sites of `newTextureWithDescriptor`? Seems kind of confusing if reading the call site, it looks like it's explicitly selecting a storage mode, but it'll be ignored. (And the fact that the `textureDescriptor` object gets modified is additional potential confusion for callers, if they re-use the object, which some do.)

Reply:
Oh good point about the Mac / macCatalyst, let me see why we had it that way. I thought it would be a driver thing (below the catalyst abstraction) but will double check. 

Also, yes will fix on Friday to set appropriately at the individual call sites.